docs/naps/8-telemetry.md

```diff
@@ -0,0 +1,181 @@
+
+The second part of this work should be setup the server to collect telemetry data. Next to collecting data, it should provide a basic public dashboard that will allow a community to see aggregated information.
+
+I propose to have the following data retention policy:
+
+1) Up to 2 weeks for logs.
+2) up 2 months of raw data (1 month of collection, then aggregation and time to validate aggregated data),
+3) infinite of aggregated data.
+  
+## Privacy assessment
+
+During the preparation of this NAP we assume that none of the collected data will be presented in 
+a form that allows to identify a single user or identify a research area of user. We also select a set of data that will be collected to minimize the possibility of reval of fragile data, but it is impossible to guarantee that it will not be possible to identify a single user (for example checking plugins combination).
+
+Because of this, we decided to not publish raw data and only show aggregated results. The aggregation will be performed using scripts. Napari core-devs will access raw data only when there will be some errors in the aggregation process.
+
+We also will publish a list of endpoints for each level of telemetry, so the given level of telemetry could be blocked on the organization level (for example by the rule on the firewall).
+
+
+If someone found that we are publishing some problematic data we will remove them and update the aggregation process to prevent such a situation in the future.
+This NAP will be updated to reflect the current state of telemetry. 
+
+ 
+## Related Work 
+
+Total systems:
+https://plausible.io/
+https://sentry.io/
+https://opentelemetry.io/
+
+Visualizations:
+https://github.com/grafana/grafana
+  
+
+  
+## Implementation 
+
+The main thing for implementation should be the low cost of maintenance. So the solution should be as simple as possible. We could either use existing solutions on the server side or implement our own.
+
+The benefit of the existing solution is that most of the work is already done. The downside is that it may require additional cost of maintenance. This cost may be caused by many features that are not needed for napari and could increase the risk of leaking data.  As I check they are implemented in techniques that are not familiar to napari core-devs. So if there will be a decision to use them we should select an SAS solution that will be maintained by the company.
+
+
+For the current, I suggest creating a simple REST API server for collecting the data. 
+It could be a simple Python FastAPI server that will store data in the SQLite database.
+
+Data for aggregation should be extracted from the database using a script running on the same machine.
+
+The output of the aggregation script should be loaded to some existing visualization tool, like grafana.
+
+It may be nice to host them on separate servers, then even if the data presented on the dashboard will be compromised, the raw data will be not exposed to the world.
```

Comment:
```suggestion
It may be nice to host raw and aggregate data on separate servers - then even if the data presented on the dashboard is compromised, the raw data will be not exposed to the world.
```